source/bdDetect.py

```diff
@@ -26,28 +26,43 @@
 from baseObject import AutoPropertyObject
 import re
 from winAPI import messageWindow
+import extensionPoints
 
 
 HID_USAGE_PAGE_BRAILLE = 0x41
 
-
 DBT_DEVNODES_CHANGED=7
 
 _driverDevices = OrderedDict()
 USB_ID_REGEX = re.compile(r"^VID_[0-9A-F]{4}&PID_[0-9A-F]{4}$", re.U)
 
-class DeviceMatch(
-	namedtuple("DeviceMatch", ("type","id", "port", "deviceInfo"))
-):
+
+class DeviceMatch(typing.NamedTuple):
 	"""Represents a detected device.
-	@ivar id: The identifier of the device.
-	@type id: str
-	@ivar port: The port that can be used by a driver to communicate with a device.
-	@type port: str
-	@ivar deviceInfo: all known information about a device.
-	@type deviceInfo: dict
 	"""
-	__slots__ = ()
+	type: str
+	"""The type of the device."""
+	id: str
+	"""The identifier of the device."""
+	port: str
+	"""The port that can be used by a driver to communicate with a device."""
+	deviceInfo: typing.Dict[str, str]
+	"""All known information about a device."""
+
+
+scanForDevices = extensionPoints.Chain[typing.Tuple[str, DeviceMatch]]()
+"""
+A Chain that can be iterated to scan for devices.
+Registered handlers should yield a tuple containing a driver name as str and DeviceMatch
+Handlers are called with these keyword arguments:
+@param Usb: Whether the handler is expected to yield USB devices.
+@type usb: bool
+@param Bluetooth: Whether the handler is expected to yield USB devices.
+@type Bluetooth: bool

```

Comment:
Fix capitalisation of parameters

```suggestion
@param usb: Whether the handler is expected to yield USB devices.
@type usb: bool
@param bluetooth: Whether the handler is expected to yield USB devices.
@type bluetooth: bool
```